bin/compile

```diff
@@ -73,16 +73,26 @@ function best_nginx_version() {
 
 function best_php_version() {
   local require_php=""
-  if [ -f "$BUILD_DIR/composer.json" ] ; then
+
+  # First check if we have something in composer.json
+  # If so, this is the value we want. It must always prevail:
+  if [ -f "${BUILD_DIR}/composer.json" ] ; then
```

Comment:
question: Could we add a warning message if `PHP_VERSION` is also specified so that customers understand which version has been chosen?

Reply:
Unfortunately we can't because this function is used with an assignment after, like so:
```bash
PHP_VERSION="$( best_php_version )"
```
So anything we `echo` in the function will be stored in the `PHP_VERSION` var :-/